sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -37,6 +37,12 @@ object StaticSQLConf {
     .stringConf
     .createWithDefault(Utils.resolveURI("spark-warehouse").toString)
 
+  val CATALOG_DEFAULT_DATABASE = buildStaticConf("spark.sql.catalog.default.database")
```

Comment:
Every catalog is free to define its own default database/namespace, see `CatalogPlugin.defaultNamespace`.

What we need here is a config to change the default database for the session catalog. How about `spark.sql.catalog.$SESSION_CATALOG_NAME.defaultDatabase`?